src/main/kotlin/com/openlattice/data/storage/PostgresEntityDataQueryService.kt

```diff
@@ -928,6 +937,41 @@ class PostgresEntityDataQueryService(
 
         return WriteEvent(tombstoneVersion, numUpdated)
     }
+
+    fun getExpiringEntitiesFromEntitySet(entitySetId: UUID, expirationBaseColumn: String, formattedDateMinusTTE: Any,
+                                         sqlFormat: Int, deleteType: DeleteType) : BasePostgresIterable<UUID> {
+        val partitionsInfo: PartitionsInfo = partitionManager.getEntitySetPartitionsInfo(entitySetId)
+        val partitions = PostgresArrays.createIntArray(hds.connection, partitionsInfo.partitions)
+        val partitionVersion = partitionsInfo.partitionsVersion
+        return BasePostgresIterable(
+                PreparedStatementHolderSupplier(
+                        hds,
+                        getExpiringEntitiesQuery(expirationBaseColumn, deleteType),
+                        FETCH_SIZE,
+                        false
+                ) {stmt ->
+                    stmt.setObject(1, entitySetId)
+                    stmt.setArray(2, partitions)
+                    stmt.setInt(3, partitionVersion)
+                    stmt.setObject(4, IdConstants.ID_ID.id)
+                    stmt.setObject(5, formattedDateMinusTTE, sqlFormat)
+                }
+        ) { rs -> ResultSetAdapters.id(rs)}
+    }
+
+    private fun getExpiringEntitiesQuery(expirationBaseColumn: String, deleteType: DeleteType) : String {
+        var ignoredClearedEntitiesClause = ""
```

Comment:
can `val` this by setting a type (pretty sure that will default it to `""`)

Reply:
oh nice

Reply:
actually seems like not :(

Reply:
`val ignoredCleardEntitiesClause: String`
Should work? Or does it not end up being empty-string?

Reply:
it does not, it ends up being an uninitialized variable

Reply:
Oooooofff. That . . . is . . . . what. But nullability whyyyyy 🤦‍♂